### ISSUE TITLE
tpm2_ptool: does not follow "password formatting standards"

### ISSUE THREAD

**Issue by maage** [tpm2-software/tpm2-pkcs11#166]
As far as I can see tool enforces to have hex encoding for some passwords, like crateprimary objauth, but not others like createprimary ownerauth.

Toos should not enforce encoding at all, or should enforce only one encoding on all passwords.

Passwords like: 'str:a', 'hex:61', 'a', '61' should be tested.

If password is 'hex:61' and encoding is not enforced, real password is 'a'. If encoding is enforced, password will fail.

If password is '61' and encoding is enforced to be hex, real password is 'a'. If it is not enforced or is enforced to be str, real password is '61'.

Quote from manuals:
```
Password Formatting
       Passwords are interpreted in  two  forms,  string  and  hex-string.   A
       string password is not interpreted, and is directly used for authoriza‐
       tion.  A hex-string, is converted from a hexidecimal form into  a  byte
       array  form, thus allowing passwords with non-printable and/or terminal
       un-friendly characters.

       By default passwords are assumed to be in the  string  form.   Password
       form is specified with special prefix values, they are:

       · str:  -  Used  to indicate it is a raw string.  Useful for escaping a
         password that starts with the “hex:” prefix.

       · hex: - Used when specifying a password in hex string format.
```

**Comment by williamcroberts** [comment]
I would just make all passwords strings. For passwords generated by the tool, they can just be max password length hex strings. Then all the code just grabs the string from the db and unwraps it with the wrapping key. This way, if someone brings in an external key with a password of "mypassword" all we have to do is wrap it. This should help reduce the amount of calls to hexlify and unhexlify.

**Comment by williamcroberts** [comment]
Essentially fixed in #300. However the issue you pointed out to in the ticket isn't quite. You can either have hex strings, hex:aabbcc or just regular strings when calling out to the tools. This is really a non issue, but now we just use regular strings everywhere and don't do this conversion.
